hfile_libcurl.c

```diff
@@ -312,6 +365,265 @@ static int add_callback_headers(hFILE_libcurl *fp) {
     return -1;
 }
 
+static int read_auth_json(auth_token *tok, hFILE *auth_fp) {
```

Comment:
This could do with a comment explaining the cryptic return values of ?, i, v and m.  The only one checked for is v apparently.